indy_node/server/domain_req_handler.py

```diff
@@ -423,18 +423,17 @@ def add_query_handler(self, txn_type, handler: Callable):
 
     def handleGetNymReq(self, request: Request):
         nym = request.operation[TARGET_NYM]
-        nymData = self.idrCache.getNym(nym, isCommitted=True)
         path = domain.make_state_path_for_nym(nym)
-        if nymData:
-            nymData[TARGET_NYM] = nym
-            data = self.stateSerializer.serialize(nymData)
-            seq_no = nymData[f.SEQ_NO.nm]
-            update_time = nymData[TXN_TIME]
-            proof = self.make_proof(path)
+        nym_data, proof = self.get_value_from_state(path, with_proof=True)
```

Comment:
So, can we remove `idrCache` at all then? Why do we need it?

Reply:
For validation, we dont query trie since its expensive to query